File: src/contests/cg/puzzle/go/the-river-i-.go
// 🎮 CodinGame Puzzle - the-river-i-
// https://www.codingame.com/training/easy/the-river-i-

package main

import (
	"bufio"
	"fmt"
	"os"
)

func digitSum(x int64) int64 {
	var s int64 = 0
	for x > 0 {
		s += x % 10
		x /= 10
	}
	return s
}

func main() {
	reader := bufio.NewReader(os.Stdin)
	var a, b int64
	fmt.Fscan(reader, &a)
	fmt.Fscan(reader, &b)
	for a != b {
		if a < b {
			a += digitSum(a)
		} else {
			b += digitSum(b)
		}
	}
	fmt.Println(a)
}

File: src/contests/cg/puzzle/perl/the-river-i-.pl
# 🎮 CodinGame Puzzle - the-river-i-
# https://www.codingame.com/training/easy/the-river-i-

sub digit_sum {
    my $x = shift;
    my $s = 0;
    while ($x > 0) {
        $s += $x % 10;
        $x = int($x / 10);
    }
    return $s;
}

my $a = <STDIN>;
chomp $a;
my $b = <STDIN>;
chomp $b;
while ($a != $b) {
    if ($a < $b) {
        $a += digit_sum($a);
    } else {
        $b += digit_sum($b);
    }
}
print "$a\n";
